R/render-table.R

```diff
@@ -3,33 +3,104 @@
+#'   (i.e. they either evaluate to \code{NA} or \code{NaN}).
+#' @param ... Arguments to be passed through to \code{\link[xtable]{xtable}}
+#'   and \code{\link[xtable]{print.xtable}}.
 #' @param env The environment in which to evaluate \code{expr}.
-#' @param quoted Is \code{expr} a quoted expression (with \code{quote()})? This
-#'   is useful if you want to save an expression in a variable.
+#' @param quoted Is \code{expr} a quoted expression (with \code{quote()})?
+#'   This is useful if you want to save an expression in a variable.
 #' @param func A function that returns an R object that can be used with
 #'   \code{\link[xtable]{xtable}} (deprecated; use \code{expr} instead).
 #'
 #' @export
-renderTable <- function(expr, ..., env=parent.frame(), quoted=FALSE, func=NULL) {
+renderTable <- function(expr, striped=FALSE, condensed=TRUE,
+                        hover=FALSE, bordered=FALSE,
+                        width="auto", align=NULL,
+                        rownames=FALSE, colnames=TRUE,
+                        digits=NULL, na="NA", ...,
+                        env=parent.frame(), quoted=FALSE, func=NULL) {
   if (!is.null(func)) {
     shinyDeprecated(msg="renderTable: argument 'func' is deprecated. Please use 'expr' instead.")
   } else {
     installExprFunction(expr, "func", env, quoted)
   }
 
+  # A small helper function to create a wrapper for an argument that was
+  # passed to renderTable()
+  createWrapper <- function(arg) {
+    if (is.function(arg)) wrapper <- reactive({ arg() })
+    else wrapper <- function() arg
+    return(wrapper)
+  }
+
+  # Create wrappers for most arguments so that functions can also be passed
+  # in, rather than only literals (useful for shiny apps)
+  stripedWrapper <- createWrapper(striped)
+  condensedWrapper <- createWrapper(condensed)
+  hoverWrapper <- createWrapper(hover)
+  borderedWrapper <- createWrapper(bordered)
+  widthWrapper <- createWrapper(width)
+  alignWrapper <- createWrapper(align)
+  rownamesWrapper <- createWrapper(rownames)
+  colnamesWrapper <- createWrapper(colnames)
+  digitsWrapper <- createWrapper(digits)
+  naWrapper <- createWrapper(na)
+
+  # Main render function
   markRenderFunction(tableOutput, function() {
-    classNames <- getOption('shiny.table.class') %OR% 'data table table-bordered table-condensed'
+    striped <- stripedWrapper()
+    condensed <- condensedWrapper()
+    hover <- hoverWrapper()
+    bordered <- borderedWrapper()
+    format <- c(striped=striped, condensed=condensed,
+                hover=hover, bordered=bordered)
+    width <- widthWrapper()
+    align <- alignWrapper()
+    rownames <- rownamesWrapper()
+    colnames <- colnamesWrapper()
+    digits <- digitsWrapper()
+    na <- naWrapper()
+
+    # For css styling
+    classNames <- paste0("table shiny-table")
+    for (i in seq_len(length(format))) {
+      if (format[i]) {
+        classNames <- paste0( classNames, " table-", names(format)[i])
+      }
+    }
```

Comment:
I'm not sure this is actually possible in here because I have to check the truth value of each element in `format` (we're only adding the class if the corresponding arg is set to `TRUE`). The closest I got is `paste0(" table-", names(format), collapse='')`, but this adds all classes and doesn't check each arg's truth value. If I try to do something like `paste0(" table-", if (format) names(format), collapse='')`, then it only checks the truth value of the first element and throws the familiar warning: `the condition has length > 1 and only the first element will be used`. Did I miss something?